cli/src/main.rs

```diff
@@ -39,40 +42,63 @@ fn main() -> Result<PipelineStatus, miette::Error> {
     }))
     .expect("miette options setup");
 
+    let mut externals = Externals::default();
+    let mut sources = SourcesCache::default();
+    let mut vm = VM::new();
+
+    build_std(&mut externals, &mut vm, &mut sources, &cli);
+
     if let Some(source) = &cli.source {
-        let name = Name::new(
-            source
-                .file_name()
-                .and_then(OsStr::to_str)
-                .expect("Incompatible filename"),
-        );
-        let mut importer = FileImporter::new({
-            let mut root = source.clone();
-            root.pop();
-            root
-        });
-        importer.add_redirection(name.clone(), source.clone());
-
-        let externals = Externals::default();
-        let mut pipeline = Pipeline::new();
-        pipeline
-            .analyzer
-            .process(name.clone(), &mut importer, &externals);
-        let diagnostics = pipeline.analyzer.take_diagnostics();
-        return Ok(use_pipeline(
-            &name,
-            SourceId(0),
-            &pipeline.analyzer,
-            &mut pipeline.vm,
-            diagnostics,
-            &mut importer,
-            &cli,
-        ));
+        return run(source, &cli, sources, externals, vm);
     }
-    let importer = FileImporter::new(
-        std::env::current_dir()
-            .into_diagnostic()
-            .context("Could not locate working directory")?,
+    let current_dir = ::std::env::current_dir()
+        .into_diagnostic()
+        .context("Could not locate working directory")?;
+
+    Ok(repl(current_dir, &cli, sources, externals, vm))
+}
+
+fn run(
+    source: &Path,
+    cli: &Cli,
+    mut sources: SourcesCache,
+    externals: Externals,
+    mut vm: VM,
+) -> Result<PipelineStatus, miette::Error> {
+    let name = Name::new(
+        source
+            .file_name()
+            .and_then(OsStr::to_str)
+            .expect("Incompatible filename"),
     );
-    Ok(prompt(importer, &cli))
+
+    let folder_path = {
+        let mut path = source.to_path_buf();
+        path.pop();
+        path
+    };
+
+    let mut importer = FileImporter::new(sources.len(), folder_path);
+
+    importer.add_redirection(name.clone(), source.to_path_buf());
+
+    let mut analyzer = Analyzer::new();
+
+    analyzer.process(name.clone(), &mut importer, &externals);
```

Comment:
```suggestion
    let mut importer = FileImporter::new(sources.len(), folder_path);
    importer.add_redirection(name.clone(), source.to_path_buf());

    let mut analyzer = Analyzer::new();
    analyzer.process(name.clone(), &mut importer, &externals);
```
They are closely related, don't separate them :cry: